File: kayak-backend/src/models/entities/workbench.rs
//! 工作台实体模型
//!
//! 定义工作台表的数据结构和相关枚举

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

/// 所有者类型枚举
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, sqlx::Type)]
#[sqlx(rename = "TEXT")]
#[sqlx(rename_all = "snake_case")]
#[serde(rename_all = "snake_case")]
pub enum OwnerType {
    /// 个人用户
    User,
    /// 团队
    Team,
}

/// 工作台状态枚举
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, sqlx::Type)]
#[sqlx(rename = "TEXT")]
#[sqlx(rename_all = "snake_case")]
#[serde(rename_all = "snake_case")]
pub enum WorkbenchStatus {
    /// 正常
    Active,
    /// 已归档
    Archived,
    /// 已删除
    Deleted,
}

impl Default for WorkbenchStatus {
    fn default() -> Self {
        WorkbenchStatus::Active
    }
}

/// 工作台实体
#[derive(Debug, Clone, FromRow, Serialize, Deserialize)]
pub struct Workbench {
    /// 工作台ID (UUID)
    pub id: Uuid,
    /// 工作台名称
    pub name: String,
    /// 描述
    pub description: Option<String>,
    /// 所有者类型
    pub owner_type: OwnerType,
    /// 所有者ID (用户ID或团队ID)
    pub owner_id: Uuid,
    /// 状态
    pub status: WorkbenchStatus,
    /// 创建时间
    pub created_at: DateTime<Utc>,
    /// 更新时间
    pub updated_at: DateTime<Utc>,
}

impl Workbench {
    /// 创建新工作台
    pub fn new(
        name: String,
        description: Option<String>,
        owner_type: OwnerType,
        owner_id: Uuid,
    ) -> Self {
        let now = Utc::now();
        Self {
            id: Uuid::new_v4(),
            name,
            description,
            owner_type,
            owner_id,
            status: WorkbenchStatus::Active,
            created_at: now,
            updated_at: now,
        }
    }
}

/// 创建工作台请求DTO
#[derive(Debug, Deserialize)]
pub struct CreateWorkbenchRequest {
    pub name: String,
    pub description: Option<String>,
    pub owner_type: OwnerType,
    pub owner_id: Uuid,
}

/// 更新工作台请求DTO
#[derive(Debug, Deserialize, Default)]
pub struct UpdateWorkbenchRequest {
    pub name: Option<String>,
    pub description: Option<String>,
    pub status: Option<WorkbenchStatus>,
}

/// 工作台响应DTO
#[derive(Debug, Serialize)]
pub struct WorkbenchResponse {
    pub id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub owner_type: OwnerType,
    pub owner_id: Uuid,
    pub status: WorkbenchStatus,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

impl From<Workbench> for WorkbenchResponse {
    fn from(wb: Workbench) -> Self {
        Self {
            id: wb.id,
            name: wb.name,
            description: wb.description,
            owner_type: wb.owner_type,
            owner_id: wb.owner_id,
            status: wb.status,
            created_at: wb.created_at,
            updated_at: wb.updated_at,
        }
    }
}

File: kayak-frontend/lib/features/workbench/models/workbench.dart
/// 工作台数据模型
///
/// 定义工作台的核心属性和序列化逻辑
library;

import 'package:freezed_annotation/freezed_annotation.dart';

part 'workbench.freezed.dart';
part 'workbench.g.dart';

/// 工作台模型
@freezed
class Workbench with _$Workbench {
  const factory Workbench({
    required String id,
    required String name,
    String? description,
    required String ownerId,
    @JsonKey(name: 'ownerType') required String ownerType,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _Workbench;

  factory Workbench.fromJson(Map<String, dynamic> json) =>
      _$WorkbenchFromJson(json);
}

/// 所有者类型枚举
enum OwnerType {
  user,
  team,
}

/// 工作台状态枚举
enum WorkbenchStatus {
  active,
  archived,
  deleted,
}

/// 分页响应模型
@freezed
class PagedWorkbenchResponse with _$PagedWorkbenchResponse {
  const factory PagedWorkbenchResponse({
    required List<Workbench> items,
    required int total,
    required int page,
    required int size,
  }) = _PagedWorkbenchResponse;

  factory PagedWorkbenchResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedWorkbenchResponseFromJson(json);
}

/// 创建工作台请求
@freezed
class CreateWorkbenchRequest with _$CreateWorkbenchRequest {
  const factory CreateWorkbenchRequest({
    required String name,
    String? description,
    @JsonKey(name: 'ownerType') String? ownerType,
  }) = _CreateWorkbenchRequest;

  factory CreateWorkbenchRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateWorkbenchRequestFromJson(json);
}

/// 更新工作台请求
@freezed
class UpdateWorkbenchRequest with _$UpdateWorkbenchRequest {
  const factory UpdateWorkbenchRequest({
    String? name,
    String? description,
  }) = _UpdateWorkbenchRequest;

  factory UpdateWorkbenchRequest.fromJson(Map<String, dynamic> json) =>
      _$UpdateWorkbenchRequestFromJson(json);
}
